Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs

```diff
@@ -61,7 +61,18 @@ public class DafnyProgramVerifier : IProgramVerifier {
           ExecutionEngine.PrintBplFile(engine.Options, fileName, boogieProgram, false, false, engine.Options.PrettyPrint);
         }
 
-        return engine.GetImplementationTasks(boogieProgram);
+        var tasksFromBoogie = engine.GetVerificationTasks(boogieProgram);
+
+        // Ordering is required to let gutter icon tests behave more deterministically
+        // In situations where there are multiple valid orders of execution
+        // In particular, GitIssue3821GutterIgnoredProblem fails without this
```

Comment:
How is that affecting that test? Normally the gutter icons should render errors on top of successful elements, so can you please give me a bit of context here? I looked at the test, and it exists because well-formedness is checked differently, and it has intermediates: false, so it should wait till everything is verified

Reply:
Seems like the sorting was no longer needed, so I've removed it and the comment.